client/extensions/woocommerce/state/wc-api/payment-methods/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import getPaymentMethodDetails from '../../../lib/get-payment-method-details';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import request from '../request';
+import { error } from '../actions';
+import {
+	WOOCOMMERCE_API_FETCH_PAYMENT_METHODS,
+	WOOCOMMERCE_API_FETCH_PAYMENT_METHODS_SUCCESS,
+} from '../../action-types';
+import {
+	arePaymentMethodsLoaded,
+	arePaymentMethodsLoading,
+} from './selectors';
+
+export const fetchPaymentMethodsSuccess = ( siteId, data ) => {
+	const paymentMethods = data.map( ( method ) => {
+		return { ...method, ...getPaymentMethodDetails( method.id ) };
+	} );
+	return {
+		type: WOOCOMMERCE_API_FETCH_PAYMENT_METHODS_SUCCESS,
+		payload: {
+			siteId,
+			data: paymentMethods,
+		},
+	};
+};
+
+export const fetchPaymentMethods = ( siteId ) => ( dispatch, getState ) => {
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( getState() );
```

Comment:
Since we have 3 calls to `getState` in this action, let just set it in a variable and pass it in.